### PR TITLE
[NL Next] Add Time-Deltas classifier

### PR DESCRIPTION
**Pull request by juliawu**
Adds a classifier for "time-delta" queries to the NL interface.
This would identify queries that ask for a change in a statvar over time, e.g. "Which places have seen the most increase in population?"

This is an initial classifier to unblock downstream tasks, improved classifiers to follow in future PRs.

Example Screenshots:
Screenshot 2023-02-01 at 3 32 43 PM

Screenshot 2023-02-01 at 3 33 48 PM

### CODE REVIEW

**Comment by pradh** [comment]
Could you add some example debug info screenshots please?

File: static/js/apps/nl_interface/debug_info.tsx [hunk 1]
```diff
@@ -166,6 +167,11 @@ export function DebugInfo(props: DebugInfoProps): JSX.Element {
               Temporal classification: {debugInfo.temporalClassification}
             </Col>
           </Row>
+          <Row>
+            <Col>
+              TimeDelta classification: {debugInfo.timeDeltaClassification}
+            </Col>
+          </Row>
```
Comment (pradh) [inline static/js/apps/nl_interface/debug_info.tsx:174]: While here, could you add debug info for comparison classification too please?

**Comment by juliawu** [comment]
> Could you add some example debug info screenshots please?

Added screenshots to the PR description!

File: server/services/nl.py [hunk 2]
```diff
@@ -192,6 +194,52 @@ def heuristic_ranking_classification(self,
...
+    """
+    subtype_map = {
+        "Increase": TimeDeltaType.INCREASE,
+        "Decrease": TimeDeltaType.DECREASE,
+    }
+
+    query = query.lower()
+    subtypes_matched = []
+    trigger_words = []
+    for subtype in constants.QUERY_CLASSIFICATION_HEURISTICS["TimeDelta"].keys(
```
Comment (pradh) [inline server/services/nl.py:220]: use a variable for this?

File: server/services/nl.py [hunk 3]
```diff
@@ -192,6 +194,52 @@ def heuristic_ranking_classification(self,
...
+    ):
+      type_trigger_words = []
+
+      for keyword in constants.QUERY_CLASSIFICATION_HEURISTICS["TimeDelta"][
+          subtype]:
+        regex = r"(^|\W)" + keyword + r"($|\W)"
```
Comment (pradh) [inline server/services/nl.py:226]: Add a comment describing this regex?

File: server/lib/nl/detection.py [hunk 4]
```diff
@@ -171,6 +177,16 @@ class CorrelationClassificationAttributes(ClassificationAttributes):
   correlation_trigger_words: str
 
 
+@dataclass
+class TimeDeltaClassificationAttributes(ClassificationAttributes):
+  """Time Delta classification attributes."""
+  time_delta_type: List[TimeDeltaType]
```
Comment (pradh) [inline server/lib/nl/detection.py:183]: nit: if its a list, use time_delta_types ?

**Review by pradh — APPROVED** [review]
Thanks!

In a follow-up PR, could we add to this too: https://github.com/datacommonsorg/website/blob/master/server/lib/nl/utterance.py#L128-L164

File: static/js/apps/nl_interface/debug_info.tsx (hunk 1 above)
Comment (pradh) [reply static/js/apps/nl_interface/debug_info.tsx:174]: Fine to defer to future PR.

Comment (juliawu) [reply static/js/apps/nl_interface/debug_info.tsx:174]: Defering to future PR

File: server/services/nl.py (hunk 2 above)
Comment (juliawu) [reply server/services/nl.py]: Switched to variable.

File: server/services/nl.py (hunk 3 above)
Comment (juliawu) [reply server/services/nl.py:227]: Added comment.

File: server/lib/nl/detection.py (hunk 4 above)
Comment (juliawu) [reply server/lib/nl/detection.py]: Renamed to time_delta_types.

**Pull request merged**